src/zapv2/__init__.py

```diff
@@ -110,25 +114,22 @@ def urlopen(self, *args, **kwargs):
            - `args`:  all non-keyword arguments.
            - `kwargs`: all other keyword arguments.
         """
-        kwargs['proxies'] = self.__proxies
-        return urllib.urlopen(*args, **kwargs).read()
+        return self.session.get(*args, proxies=self.__proxies, **kwargs).text
 
-    def _request_api(self, url, get=None):
+    def _request_api(self, url, query=None):
         """
         Shortcut for an API request. Will always add the apikey (if defined)
 
         :Parameters:
            - `url`: the url to GET at.
         """
-        session = requests.Session()
-        if self.__apikey is not None:
```

Comment:
I moved session into instance property because there's no point in calling during every request.